portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,31 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+/*
+* Scrolls to top of element with id "top"- top of page. Adjusts scroll position to account for
+* height for header/navigation bar
+*/
+function scrollToID(id) {
+  var y = document.getElementById(id).offsetTop - document.getElementById("header").offsetHeight;
+  //console.log("height of header " + document.getElementById("header").offsetHeight);
+  console.log("top of elm " + document.getElementById(id).offsetTop);
+  //console.log("jumping to " + y);
+  window.scrollTo({top: y, behavior: 'smooth'});
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+window.onscroll = function() {
+    highlightTab();
+};
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
+/*
+* Highlights current tab on nav bar
+*/
+function highlightTab() {
+    if (window.pageYOffset < document.getElementById("project").offsetTop - document.getElementById("header").offsetHeight) {
+        document.getElementById("about-btn").classList.add("curr-selection");
```

Comment:
Fetching elements every time this func is called (a lot because it's a scroll listener) will make the browser do more work than it needs to. Consider querying once and using those variables in the function.

```js
var projectElem = document.getElementById("project");
function highlightTab() {
    if (window.pageYOffset < projectElem.offsetTop ...) {...}
    ...
}
```

Reply:
+1 to querying once.

One gotcha might run into when doing Gabriel's suggestion is document.getElementById("project") will return null as shown because your script.js is at the top of the page and the browser will execute that first before it creates the dom for the html below.

If you run into this, I would suggestion moving the script tag to the bottom of the body tag below all the html.